src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1221,6 +1215,38 @@ public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
             }
         }
 
+        private ImmutableArray<MethodSymbol> ExplicitlyOverriddenMethods()
+        {
+            var moduleSymbol = _containingType.ContainingPEModule;
+
+            // Context: we need the containing type of this method as context so that we can substitute appropriately into
+            // any generic interfaces that we might be explicitly implementing.  There is no reason to pass in the method
+            // context, however, because any method type parameters will belong to the implemented (i.e. interface) method,
+            // which we do not yet know.
+            return new MetadataDecoder(moduleSymbol, _containingType).GetExplicitlyOverriddenMethods(_containingType.Handle, _handle, this.ContainingType);
+        }
+
+        internal MethodSymbol ExplicitlyOverriddenClassMethod
+        {
+            get
+            {
+                if (IsExplicitClassOverride)
+                {
+                    foreach (MethodSymbol method in ExplicitlyOverriddenMethods())
```

Comment:
>ExplicitlyOverriddenMethods [](start = 52, length = 27)

I think it would be good to test scenarios when we fail to successfully resolve a MethodImpl. For example, the method is not found in the type, or the type is not found. Cover scenarios when the type is a class and an interface. Also in presence of other resolvable MethodImpls, from classes and interfaces, all combinations.

Reply:
This suggestion is tracked at [https://github.com/dotnet/roslyn/issues/44079](https://github.com/dotnet/roslyn/issues/44079)

---
In reply to: [422233060](https://github.com/dotnet/roslyn/pull/43797#discussion_r422233060) [](ancestors = 422233060)